src/Core/Resolvers/CosmosMutationEngine.cs

```diff
@@ -233,6 +235,14 @@ private static async Task<ItemResponse<JObject>> HandleUpdateAsync(IDictionary<s
             }
         }
 
+        private void CheckReadAccess(string entityName)
+        {
+            if (!_authorizationResolver.AreRoleAndOperationDefinedForEntity(entityName, AuthorizationType.Anonymous.ToString(), EntityActionOperation.Read))
+            {
+                throw new Exception("The database operation was successful but the current user is unable to view the response due to lack of read permissions");
+            }
+        }
```

Comment:
Why hardcoding to `AuthorizationType.Anonymous`? 

You should be acquiring the current role from the GraphQL execution context. See `AuthorizeMutationFields(...)` which has a snippet that you can use.